framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java

```diff
@@ -73,15 +73,16 @@ public static void annotateSupers(
      */
     @Override
     public int getElementIndex() {
-        return index;
+        return index == 0xffff ? -1 : index;
```

Comment:
Where is the index field set? It would be better to have this check where `TypeAnnotationPosition.type_index` is read, to see why this is needed.